src/contracts/strategies/StrategyBase.sol

```diff
@@ -223,8 +223,8 @@ contract StrategyBase is Initializable, Pausable, IStrategy {
      * @return The amount of underlying tokens corresponding to the input `amountShares`
      * @dev Implementation for these functions in particular may vary significantly for different strategies
      */
-    function sharesToUnderlying(uint256 amountShares) public view virtual override returns (uint256) {
-        return sharesToUnderlyingView(amountShares);
+    function sharesToUnderlying(uint256 amountShares) public virtual override returns (uint256 underlying) {
+        emit SharesToUnderlying(underlying = sharesToUnderlyingView(amountShares));
```

Comment:
it should be the ratio, not total underlying token. The ratio can be by setting amountShares = 10^8

besides, felt like emitting total shares is also super useful and emitting both would actually be better? @ypatil12 wdyt?

Reply:
Ahh gotcha. Okay let's emit the ratio